File: bionic-js-ios/BjsTests/Bjs/UserOfToyComponent1.swift
import JavaScriptCore
import Bjs

class UserOfToyComponent1: BjsObject {
    
    static var lastToy: ToyComponent1? {
        get {
            return Bjs.get.getWrapped(Bjs.get.getProperty(UserOfToyComponent1.bjsClass, "lastToy"), ToyComponent1.self)
        }
        set {
            Bjs.get.setProperty(UserOfToyComponent1.bjsClass, "lastToy", Bjs.get.putWrapped(newValue, ToyComponent1Wrapper.self))
        }
    }
    
    class func add(_ offset: Int?, _ int1: Int?, _ int2: Int?) -> Int? {
        return Bjs.get.getInt(Bjs.get.call(UserOfToyComponent1.bjsClass, "add", Bjs.get.putPrimitive(offset),
                                           Bjs.get.putPrimitive(int1), Bjs.get.putPrimitive(int2)))
    }
    
    class func getToy(_ int1: Int?, _ int2: Int?) -> ToyComponent1? {
        return Bjs.get.getWrapped(Bjs.get.call(UserOfToyComponent1.bjsClass, "getToy", Bjs.get.putPrimitive(int1),
                                              Bjs.get.putPrimitive(int2)), ToyComponent1.self)
    }
    
    class func getSum(_ toy1: ToyComponent1?, _ toy2: ToyComponent1?) -> Int? {
        return Bjs.get.getInt(Bjs.get.call(UserOfToyComponent1.bjsClass, "getSum",
                                           Bjs.get.putWrapped(toy1, ToyComponent1Wrapper.self),
                                           Bjs.get.putWrapped(toy2, ToyComponent1Wrapper.self)))
    }
    
    class func isLastToy(_ toy: ToyComponent1?) -> Bool? {
        return Bjs.get.getBool(Bjs.get.call(UserOfToyComponent1.bjsClass, "isLastToy",
                                           Bjs.get.putWrapped(toy, ToyComponent1Wrapper.self)))
    }
    
    
    override class var bjsModulePath: String {
        return "/src/UserOfToyComponent1"
    }
    
    class func bjsFactory(_ jsObject: JSValue) -> UserOfToyComponent1 {
        return UserOfToyComponent1(jsObject)
    }
}

File: bionic-js-ios/BjsTests/Bjs/test.bundle/src/UserOfToyComponent1.js
const {ToyComponent1} = require('./ToyComponent1')

class UserOfToyComponent1 {
    
    // @bionic ToyComponent1
    static get lastToy() {
        return this._lastToy
    }
    
    // @bionic ToyComponent1
    static set lastToy(lastToy) {
        this._lastToy = lastToy
    }
    
    // @bionic (ToyComponent1) => Bool
    static isLastToy(toy) {
        return toy === this._lastToy
    }
    
    // @bionic (Int, Int, Int) => Void
    static add(offset, int1, int2) {
        const toy = new ToyComponent1(int1, int2)
        return toy.getSum(offset)
    }
    
    // @bionic (Int, Int) => ToyComponent1
    static getToy(int1, int2) {
        this.lastToy = new ToyComponent1(`${int1}`, `${int2}`)
        return this.lastToy
    }
    
    // @bionic (ToyComponent1, ToyComponent1) => Int
    static getSum(toy1, toy2) {
        return toy1.getToySum(toy2)
    }
}

module.exports = {UserOfToyComponent1}
